framework/include/actions/AddLikelihoodAction.h

```diff
@@ -0,0 +1,25 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseObjectAction.h"
+
+/**
+ * This class adds a likelihood object.
+ */
+class AddLikelihoodAction : public MooseObjectAction
```

Comment:
I am against this, this action and everything relate to `Likelihood` should go to the STM. You can follow the example of Covariancefunction.